sound/soc/sof/amd/acp.c

```diff
@@ -529,6 +547,19 @@ int amd_sof_acp_probe(struct snd_sof_dev *sdev)
 	sdev->debug_box.size = BOX_SIZE_1024;
 
 	adata->signed_fw_image = false;
+	dmi_id = dmi_first_match(acp_sof_quirk_table);
+	if (dmi_id && dmi_id->driver_data) {
+		adata->fw_code_bin = kasprintf(GFP_KERNEL, "%s/sof-%s-code.bin",
+					       plat_data->fw_filename_prefix,
+					       chip->name);
+		adata->fw_data_bin = kasprintf(GFP_KERNEL, "%s/sof-%s-data.bin",
+					       plat_data->fw_filename_prefix,
+					       chip->name);
```

Comment:
need to test that kasprintf worked....

Reply:
Yes, It is working fine, debug log attached below
[    2.923341] snd_sof_amd_vangogh 0000:04:00.5: fw_code_bin:amd/sof/sof-vangogh-code.bin, fw_data_bin:amd/sof/sof-vangogh-data.bin